src/main/java/com/ericsson/ei/subscription/RunSubscription.java

```diff
@@ -52,6 +55,8 @@
     @Getter
     @Value("${spring.data.mongodb.database}")
     public String dataBaseName;
+    
+    Map<String, List<String>> subscriptionsCache = new HashedMap<>();
```

Comment:
Is this subscriptionCache static/volatile memory initiated for all threads memories? So same Hashmap object is used in all threads.
If not, then I think its possible to get parallel/several Subscription notifications when only one subscription notification should be made.

Reply:
No, we only write to this inside a sync block where we go to the database to check. This is only to reduce the number of checks towards the database.

Reply:
So one thread process the Subscription matching per aggregations and matching subscription.

So we can have lock state here if the Synch thread is not releasing the code to be executed by next thread and it might cause slower pace for matching and doing notification requests.